src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilder.cs

```diff
@@ -144,6 +153,28 @@ public ClusterBuilder ConfigureConnectionPool(Func<ConnectionPoolSettings, Conne
             _connectionPoolSettings = configurator(_connectionPoolSettings);
             return this;
         }
+        
+        /// <summary>
+        /// Configures the SDAM logging settings.
+        /// </summary>
+        /// <param name="configurator">The SDAM logging settings configurator delegate.</param>
+        /// <returns>A reconfigured cluster builder.</returns>
+        public ClusterBuilder ConfigureSdamLogging(Func<SdamLoggingSettings, SdamLoggingSettings> configurator)
+        {
+            _sdamLoggingSettings = configurator(_sdamLoggingSettings);
+            if (!_sdamLoggingSettings.LoggingEnabled)
+            {
+                return this;
+            }
+            var traceSource = new TraceSource(__traceSourceName, SourceLevels.All);
+            traceSource.Listeners.Clear(); // remove the default listener
+            var listener = _sdamLoggingSettings.LogToStdout
+                ? new TextWriterTraceListener(Console.Out)
+                : new TextWriterTraceListener(new FileStream(_sdamLoggingSettings.LogFilename, FileMode.Append));
```

Comment:
We could use a string fileName here if we were on a newer version TraceSource/TextWriterTraceListener, but to avoid upgrading dependencies and to avoid using ifdefs, we're using an older constructor available in .NET Framework 4.5 and the backported 4.0 packages.